documentation/internal/Element-Type-Specs.md

```diff
@@ -33,20 +33,53 @@ _Free-form, single-line text input element_
+- **label**: `string` -- Text that shows in the HTML "label" attribute of the form element (Markdown string, with dynamic expression evaluation)
 - **description**: `string` -- additional explanatory text (usually not required) [Optional]
 - **placeholder**: `string`-- text to display before user input (HTML "placeholder" attribute) [Optional]
 - **maskedInput**: `boolean` -- if `true`, displays user input as masked (hidden) characters -- i.e. for passwords. [Optional]
-- **minWidth**/**maxWidth**: `integer` -- optional controls over visual display [Optional] _(We may never use these)_
+- ~~**minWidth**/**maxWidth**: `integer` -- optional controls over visual display [Optional]~~ _(Not currently implemented -- We may never use these)_
 
 #### Response type
 
-_This describes the expected object that will be stored in the `application_response` table `value` field. from the user's response_  
+_This describes the expected object that will be stored in the `application_response` table `value` field from the user's response_  
 `{ text: <string> }`
 
 ---
 
-### Text Information _(not yet implemented)_
+### Password Input
+
+- **type/code**: `password`
+- **category**: `Question`
+
+_Secure password input field, with enter-twice confirmation_
+
+#### Input parameters (in the `parameters` JSON)
+
+- **label**: `string` -- Text that shows in the HTML "label" attribute of the form element (Markdown string, with dynamic expression evaluation)
+- **description**: `string` -- additional explanatory text (usually not required) [Optional]
+- **placeholder**: `string`-- text to display before user input (HTML "placeholder" attribute) [Optional]
+- **confirmPlaceholder**: `string` -- text to display as placeholder in the password confirm input field [Default: `"Confirm password"`]
+- **maskedInput**: `boolean` -- if `true`, displays user input as masked (hidden) characters -- i.e. for passwords. [Optional -- default `true`]
+- **showPasswordToggle**: `boolean` -- if `true`, displays a checkbox to show the masked input as regular text [Optional -- default `true`]
+- **validationInternal**: `JSON` -- a dynamic expression for checking if the user's response is a valid input. The validation method must be internal to the plugin (as opposed to the general "validation" field on all plugins) since the password is never exposed to the Wrapper, only the hash, so any checks on its validity must take place in the plugin's own validation method.
+- **validationMessageInternal**: `string` -- the message that shows in the UI when validation fails. (For internal validation, as above)
+
+Note that, as the password itself is never stored (only the hash), this fields input is reset if the page is reloaded.
+
+#### Response type
+
+_This describes the expected object that will be stored in the `application_response` table `value` field from the user's response_
+
+```
+{
+  text: <string> (Only used to display '•••••••' on the summary page)
```

Comment:
What is this text storing? Just the number of characters in \* ?

Reply:
Yes, it's just '•••••••' regardless, not even dependent on the number of characters (which is what it should be, since not even the number of chars should be stored outside the plugin).

It serves two purposes:
1. Something to display on the Summary page
2. The revalidateAll method checks the `text` field of the Response and if blank considers that a fail, so this ensures it will pass that check.

Reply:
Alright! Thanks, I was just curious